repo/fsrepo/migrations/retryfetcher.go

```diff
@@ -0,0 +1,37 @@
+package migrations
+
+import (
+	"context"
+	"fmt"
+)
+
+type RetryFetcher struct {
+	Fetcher
+	maxRetries int
+}
+
+var _ Fetcher = (*RetryFetcher)(nil)
+
+func NewRetryFetcher(baseFetcher Fetcher, maxRetries int) *RetryFetcher {
+	return &RetryFetcher{Fetcher: baseFetcher, maxRetries: maxRetries}
+}
+
+func (r *RetryFetcher) Fetch(ctx context.Context, filePath string) ([]byte, error) {
+	var lastErr error
+	for i := 0; i < r.maxRetries; i++ {
+		out, err := r.Fetcher.Fetch(ctx, filePath)
```

Comment:
The RetryFetcher made me reconsider just returning []byte with the interface change, WDYT? Seems like having writers is sort of useless with the retryFetcher until we have a `FetchRange` function or allow passing a range into `Fetch` since we'd otherwise have to buffer it to prevent the Writer from getting corrupted.

We can revert the commit that did this and use writers again, but until we add range support it might not be worth it and it's not like breaking this interface is that costly for us

Reply:
another option is to pass the destination file path into the fetcher, then we wouldn't need to buffer in-mem (write to tmp, then move if/when it succeeds)

Reply:
We're not always using tmp files here (e.g. when we download the version file we're happy to just keep the whole thing in memory). Passing in an FD also seems more complicated since you need to worry about similar types of overwriting as with a WriterSeeker.

It's a bit annoying to need to keep the whole file in memory, but hopefully this shouldn't be too big a deal since it's happening before daemon startup when most other resource usage is lower anyway.